src/server/scripts/OutdoorPvP/OutdoorPvPNA.cpp

```diff
@@ -664,18 +856,55 @@ void OPvPCapturePointNA::UpdateWyvernRoostWorldState(uint32 roost)
 class OutdoorPvP_nagrand : public OutdoorPvPScript
 {
 public:
-    OutdoorPvP_nagrand()
-        : OutdoorPvPScript("outdoorpvp_na")
-    {
-    }
+    OutdoorPvP_nagrand() : OutdoorPvPScript("outdoorpvp_na") { }
 
     OutdoorPvP* GetOutdoorPvP() const override
     {
         return new OutdoorPvPNA();
     }
 };
 
+struct outdoorpvp_na_halaa_creatures : public ScriptedAI
+{
+    outdoorpvp_na_halaa_creatures(Creature* creature) : ScriptedAI(creature) { }
+
+    void UpdateAI(uint32 diff)
+    {
+        if(halaaNPCHorde.size() != NA_HALAA_CREATURE_TEAM_SPAWN && halaaNPCAlly.size() != NA_HALAA_CREATURE_TEAM_SPAWN)
+        {
+            std::list<Creature*> creatures;
+            uint32 entry = 0;
+            for (int i = 0; i < NA_HALAA_CREATURES; i++)
+            {
+                me->GetCreatureListWithEntryInGrid(creatures, PatrolCreatureEntry[i].idPatrol, 250);
+            }
+
+            if (creatures.size() == NA_HALAA_MAX_CREATURE_SPAWN)
+            {
+                for (std::list<Creature*>::iterator itr = creatures.begin(); itr != creatures.end(); ++itr)
+                {
+                    Creature* const c = *itr;
+                    if (entry < NA_HALAA_CREATURE_TEAM_SPAWN)
+                    {
+                        halaaNPCHorde[entry] = (c->GetSpawnId());
+                    }
+                    else
+                    {
+                        halaaNPCAlly[entry - NA_HALAA_CREATURE_TEAM_SPAWN] = (c->GetSpawnId());
+                    }
+                    c->AddObjectToRemoveList();
+                    entry++;
+                    sObjectMgr->RemoveCreatureFromGrid(c->GetSpawnId(), c->GetCreatureData());
+                }
+            }
+        }
+        DoMeleeAttackIfReady();
+    }
+    void Reset() override { }
```

Comment:
is this really needed when empty

Reply:
I was testing something but after that i forgot to delete